src/wp-includes/blocks.php

```diff
@@ -2420,6 +2421,52 @@ function build_query_vars_from_query_block( $block, $page ) {
 				}
 			}
 		}
+		if ( ! empty( $block->context['query']['format'] ) && is_array( $block->context['query']['format'] ) ) {
+			$formats   = $block->context['query']['format'];
```

Comment:
Validation: Ensure that the formats come from the supported values only provided by `get_post_format_slugs()`.

⚠️ Untested but I think you want something along the lines of:

```php
$formats = array_intersect( $formats, get_post_format_slugs() );
```

Reply:
Though could I add this after the check for the standard format, since format is not (usually) in get_post_format_slugs()?